### PR TITLE
Fix hl assets example

### LINKED ISSUE #609: amethyst_assets hl example does not run as expected
# Problem

I cannot run the example in amethyst_assets/examples/hl.rs using the standard `cargo run --example hl` method. If there is an extra bit to this command to get it to run then it is not documented in an obvious location.

# Expected

for the hl example to work like the ll example 
```
aaron@neptune:~/Programming/amethyst/amethyst_assets$ cargo run --example ll --features profile
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `/home/aaron/Programming/amethyst/target/debug/examples/ll`
dummy: Some(DummyAsset(">> Hello World!!\n"))
aaron@neptune:~/Programming/amethyst/amethyst_assets$ 
```

# Example

```aaron@neptune:~/Programming/amethyst/amethyst_assets$ cargo run --example hl
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `/home/aaron/Programming/amethyst/target/debug/examples/hl`
thread '<unnamed>' panicked at 'No resource with the given id', libcore/option.rs:917:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
aaron@neptune:~/Programming/amethyst/amethyst_assets$
 ```

## Edits
Fixed some text inside the first in-line code block

### PR DESCRIPTION
**Pull request by torkleyy**
Fixes #609 

cc @zoonage

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/616)

### CODE REVIEW

File: amethyst_assets/examples/hl.rs [hunk 1]
```diff
@@ -36,6 +37,7 @@ impl App {
         world.add_resource(Errors::new());
         world.add_resource(AssetStorage::<MeshAsset>::new());
         world.add_resource(Loader::new(path, pool.clone()));
+        world.add_resource(Time::placeholder());
```
Comment (zoonage) [inline amethyst_assets/examples/hl.rs]: Could this use `Time::default` instead of creating a new `Time::placeholder` method?

Comment (torkleyy) [reply amethyst_assets/examples/hl.rs]: Much better, I didn't see the `Default` implementation.

File: amethyst_assets/examples/hl.rs [hunk 2]
```diff
@@ -177,6 +190,7 @@ fn main() {
         .add(RenderingSystem, "rendering", &[])
         .build();
 
-    let mut app = App::new(disp, "examples/assets/", State::Start);
+    let assets_dir = format!("{}/examples/assets/", env!("CARGO_MANIFEST_DIR"));
```
Comment (zoonage) [inline amethyst_assets/examples/hl.rs]: I think this line will break on windows builds due to how it expects the path to be formatted. Maybe something like

```
let assets_dir = if cfg!(windows) {
    format!("{}\examples\assets\", env!("CARGO_MANIFEST_DIR"))
} else {
    format!("{}/examples/assets/", env!("CARGO_MANIFEST_DIR"))
};
```

Shamelessly copied from https://doc.rust-lang.org/std/macro.cfg.html

**Comment by torkleyy** [comment]
I'm fairly certain slashes also work on Windows here.

**Comment by zoonage** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[amethyst_assets/examples/hl.rs, line 193 at r2](https://beta.reviewable.io/reviews/amethyst/amethyst/616#-L8uiygsAK0-5S0BtTpl-r2-193:-L8vKOie1Ih8-BsmuIrE:b32w0x) ([raw file](https://github.com/amethyst/amethyst/blob/efbfdee4332402e4d77a611f109aa5c691b0426c/amethyst_assets/examples/hl.rs#L193)):*
<details><summary><i>Previously, zoonage (Aaron George) wrote…</i></summary><blockquote>

I think this line will break on windows builds due to how it expects the path to be formatted. Maybe something like

```
let assets_dir = if cfg!(windows) {
    format!("{}\examples\assets\", env!("CARGO_MANIFEST_DIR"))
} else {
    format!("{}/examples/assets/", env!("CARGO_MANIFEST_DIR"))
};
```

Shamelessly copied from https://doc.rust-lang.org/std/macro.cfg.html
</blockquote></details>

OK

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/amethyst/amethyst/616#-:-L8vLOha7mHadjOpvZ4K:bnfp4nl)*

File: amethyst_assets/examples/hl.rs (hunk 2 above)
Comment (Rhuagh) [reply amethyst_assets/examples/hl.rs]: I believe rust stod handles it.

Comment (jojolepro) [reply amethyst_assets/examples/hl.rs]: There won't be issues from forward slashes normally.

File: amethyst_assets/examples/hl.rs [hunk 3]
```diff
@@ -148,10 +149,22 @@ impl State {
...
                 Some(State::Loading(progress))
             }
-            State::Loading(progress) => if progress.is_complete() {
-                Some(State::SomethingElse)
-            } else {
-                Some(State::Loading(progress))
+            State::Loading(progress) => match progress.complete() {
+                Completion::Complete => Some(State::SomethingElse),
+                Completion::Failed => {
+                    eprintln!("Asset loading failed!");
```
Comment (jojolepro) [inline amethyst_assets/examples/hl.rs]: Could you use the log crate?
error!("..")

Comment (torkleyy) [reply amethyst_assets/examples/hl.rs]: No, there's no logging in these examples. Not necessary here.

**Pull request merged**